### PR TITLE
Fix GCE ILB code to support protocol change

### PR DESCRIPTION
**Pull request by prameshj**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

This bug was introduced when adding support for Global Access - https://github.com/kubernetes/kubernetes/pull/81549
The forwarding rule deletion logic was refactored, which will now error out if ILB service protocol is changed. This PR fixes the behavior and adds a unit test.

**Which issue(s) this PR fixes**:

Fixes # 93422

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go [hunk 1]
```diff
@@ -200,10 +194,31 @@ func (g *Cloud) ensureInternalLoadBalancer(clusterName, clusterID string, svc *v
 	if options.AllowGlobalAccess {
 		newFwdRule.AllowGlobalAccess = options.AllowGlobalAccess
 	}
-	if err := g.ensureInternalForwardingRule(existingFwdRule, newFwdRule); err != nil {
+
+	fwdRuleDeleted := false
+	if existingFwdRule != nil && !forwardingRulesEqual(existingFwdRule, newFwdRule) {
```
Comment (freehan) [inline staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go:199]: I have some concern regarding this line. If there is any bug in the logic generating newFwdRule, this is likely to cause a delete/create loop for the forwarding rules.

The bug can potentially due to 1. a new version of controller added a new field 2. GCE defaulting behavior change etc

Feels a bit risky to compare the whole forwarding rule. Does it make sense to do spot checks? For instance, check the protocol and port and etc....

Comment (freehan) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go:199]: Not sure how was this handled before?

Comment (prameshj) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go:199]: This is the same code from before - https://github.com/kubernetes/kubernetes/blob/a34f1e483104bd51c3e9a6aec3dbbcf6301789da/staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go#L143

Comment (prameshj) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go:199]: forwardingRulesEqual compares select fields - https://github.com/kubernetes/kubernetes/blob/a34f1e483104bd51c3e9a6aec3dbbcf6301789da/staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go#L750

**Review by freehan — APPROVED** [review]
/lgtm
/approve

**Comment by prameshj** [comment]
Thanks @justaugustus !

**Pull request merged**